### ISSUE TITLE
TypeError: 'NoneType' object is not subscriptable

### ISSUE THREAD

**Issue by sohsatoh** [codsane/autotss#2]
Checking devices.ini for new devices...
Device: [iPhone X] ECID: [******] Board Config: [d22ap]
Added 0 new devices to the database

Saving unsaved blobs for 1 devices...
Traceback (most recent call last):
  File "autotss.py", line 190, in <module>
    main()
  File "autotss.py", line 187, in main
    autotss()
  File "autotss.py", line 21, in __init__
    self.checkAllDevices()
  File "autotss.py", line 115, in checkAllDevices
    for firmware in self.liveFirmwareAPI[device['deviceID']]['firmwares']:
TypeError: 'NoneType' object is not subscriptable

Could you fix it?

**Comment by codsane** [comment]
> Device: [iPhone X] ECID: [******] Board Config: [d22ap]

Double check the value you are using for the device identifier in your devices.ini file. I have noticed that you have used "iPhone X" as your device name, but keep in mind that "iPhone X" cannot be used as the value for the device identifier. Device identifiers follow a specific format and can be found [here](https://www.theiphonewiki.com/wiki/Models). Your device identifier should be either "iPhone10,3" or "iPhone10,6".

**Comment by sohsatoh** [comment]
Thank you for the reply.

devices.ini is maybe correct.
```
[iPhoneX]
identifier = iPhone10,3 
ecid = D38913xxxxxxE
boardconfig = d22ap
```

but I still got error.
```
Traceback (most recent call last):
  File "autotss.py", line 190, in <module>
    main()
  File "autotss.py", line 187, in main
    autotss()
  File "autotss.py", line 21, in __init__
    self.checkAllDevices()
  File "autotss.py", line 115, in checkAllDevices
    for firmware in self.liveFirmwareAPI[device['deviceID']]['firmwares']:
TypeError: 'NoneType' object is not subscriptable
```

**Comment by sohsatoh** [comment]
Sorry, it is my fault.

I mistakenly delete this lines.
```
[Device Name]
identifier = iPhone9,4
ecid = 1438617935153
```

Thank you for your support.

**Comment by sohsatoh** [comment]
Sorry for reopening the issue.

I wrote device.ini like this.
```
[iPhoneX]
identifier = iPhone10,3
ecid = D389138xxxxxE

[iPhoneX2]
identifier = iPhone10,3
ecid = D389138xxxxxE
boardconfig = d22ap
```

After running command, it says "Done saving blobs".
However, in blobs/iPhone10,3/D389138xxxxxE/release/11.2, there is no shsh2 file.

**Comment by sohsatoh** [comment]
I sent pull request.
Thank you.

**Comment by codsane** [comment]
Do you have a copy of your config.ini that was giving you issues that you could attach? Feel free to change up the ECID a bit if you need to.

And yes, I have noticed the pull request. Did this fix all the problems you were having early?

**Comment by sohsatoh** [comment]
I found that when I write like
```
[iPhoneX2]
identifier = iPhone10,3
ecid = D389138xxxxxE
boardconfig = d22ap
```
TypeError: 'NoneType' object is not subscriptable.

When I write config.ini like
```
[iPhoneX]
identifier = iPhone10,3
ecid = D389138xxxxxE

[iPhoneX2]
identifier = iPhone10,3
ecid = D389138xxxxxE
boardconfig = d22ap
```
success to run script.

So, what i wanna say is 
Somehow the script requires writing 2 or more devices' information in config.ini.


Anyways, the pull request fixes the problem that shsh2 can not be saved.
It would not fix the problem above.

**Comment by codsane** [comment]
I temporarily merged your code to fix some of the iPhone X problems. Going to take a look into the others.

**Comment by codsane** [comment]
The TypeError should be fixed in the code I pushed. There was a bug where the IPSW.me API wasn't being grabbed when there was already a board config specified in the config file (thanks @leftyfl1p). 

I've tried your devices.ini and haven't had any issues, except for iOS 11.2 Build 15C113. For some reason this specific build doesn't play nice with tsschecker, however iOS 11.2 Build 15C114 has no issues.

I had no idea there were multiple iOS builds with the same version number. Because of this, blobs are now stored in `/blobs/device model/device ecid/version number/version build`.

The script will now check to make sure the blob files are actually there, and if they aren't, a log file will be saved in place of the blob file. Give the new version a shot and if you're still having issues (outside of iOS 11.2 Build 15C113), post the tsschecker_log.txt files.
